ccl/oidcccl: TestOIDCAuthorization_UserinfoPaths failed

ccl/oidcccl.TestOIDCAuthorization_UserinfoPaths [failed](https://mesolite.cluster.engflow.com/invocations/default/81761ce7-6a95-46c6-8f6c-957b0b55b178?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL29pZGNjY2w6b2lkY2NjbF90ZXN0) on release-26.1 @ [f6c86c46137cfaedcb02b8409f9dba99bf8bede9](https://github.com/cockroachdb/cockroach/commits/f6c86c46137cfaedcb02b8409f9dba99bf8bede9):


```
=== RUN   TestOIDCAuthorization_UserinfoPaths
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestOIDCAuthorization_UserinfoPaths1544812280
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    authorization_oidc_test.go:578: -- test log scope end --
test logs left over in: outputs.zip/logTestOIDCAuthorization_UserinfoPaths1544812280
--- FAIL: TestOIDCAuthorization_UserinfoPaths (159.27s)
=== RUN   TestOIDCAuthorization_UserinfoPaths/userinfo_success
    authorization_oidc_test.go:560: 
        	Error Trace:	pkg/ccl/oidcccl/authorization_oidc_test.go:560
        	Error:      	Received unexpected error:
        	            	Get "https://127.0.0.1:46815/oidc/v1/callback?code=dummy&state=GiAMU6pwGp-LjZH33W_hk6vc3Wd5XijA5bOWTs8N-7z71iIgyirq7n44rUfSzxvNPwd8rgnrz-F7Uf72WDKPSdIVlwg%3D": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
        	Test:       	TestOIDCAuthorization_UserinfoPaths/userinfo_success
--- FAIL: TestOIDCAuthorization_UserinfoPaths/userinfo_success (11.21s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/product-security
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOIDCAuthorization_UserinfoPaths.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63579

--- a/pkg/security/oidcauth/BUILD.bazel
+++ b/pkg/security/oidcauth/BUILD.bazel
@@ -40,6 +40,7 @@ go_library(
 
 go_test(
     name = "oidcauth_test",
+    size = "medium",
     srcs = [
         "authentication_oidc_test.go",
         "authorization_oidc_test.go",

--- a/pkg/security/oidcauth/authorization_oidc_test.go
+++ b/pkg/security/oidcauth/authorization_oidc_test.go
@@ -309,6 +309,9 @@ func TestOIDCAuthorization_TokenPaths(t *testing.T) {
 			rpcCtx := app.NewClientRPCContext(ctx, username.TestUserName())
 			client, err := rpcCtx.GetHTTPClient()
 			require.NoError(t, err)
+			// The default 10s timeout is too short under race detection where OIDC
+			// callback processing can take 20s+.
+			client.Timeout = 45 * time.Second
 			// Prevent automatic redirects so we can capture cookies and headers.
 			client.CheckRedirect = func(*http.Request, []*http.Request) error { return http.ErrUseLastResponse }
 
@@ -385,6 +388,9 @@ func TestOIDCAuthorization_UserinfoPaths(t *testing.T) {
 	rpc := app.NewClientRPCContext(ctx, username.TestUserName())
 	cl, err := rpc.GetHTTPClient()
 	require.NoError(t, err)
+	// The default 10s timeout is too short under race detection where OIDC
+	// callback processing can take 20s+.
+	cl.Timeout = 45 * time.Second
 	cl.CheckRedirect = func(*http.Request, []*http.Request) error { return http.ErrUseLastResponse }
 
 	// Create an RSA key pair for signing and verifying tokens.
@@ -738,6 +744,9 @@ func TestOIDCAuthorization_RoleGrantAndRevoke(t *testing.T) {
 	rpcCtx := app.NewClientRPCContext(ctx, username.TestUserName())
 	client, err := rpcCtx.GetHTTPClient()
 	require.NoError(t, err)
+	// The default 10s timeout is too short under race detection where OIDC
+	// callback processing can take 20s+.
+	client.Timeout = 45 * time.Second
 	// Prevent automatic redirects so we can capture cookies and headers.
 	client.CheckRedirect = func(*http.Request, []*http.Request) error { return http.ErrUseLastResponse }
 

